tests/net/ethernet_mgmt/src/main.c

```diff
@@ -0,0 +1,283 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <net/net_if.h>
+#include <net/ethernet.h>
+#include <net/ethernet_mgmt.h>
+
+#include <ztest.h>
+
+static const u8_t mac_addr_init[6] = { 0x01, 0x02, 0x03,
+				       0x04,  0x05,  0x06 };
+
+static const u8_t mac_addr_change[6] = { 0x01, 0x02, 0x03,
+					 0x04,  0x05,  0x07 };
+
+struct eth_fake_context {
+	struct net_if *iface;
+	u8_t mac_address[6];
+
+	bool auto_negociation;
+	bool full_duplex;
+	bool link_10bt;
+	bool link_100bt;
+};
+
+static struct eth_fake_context eth_fake_data;
+
+static void eth_fake_iface_init(struct net_if *iface)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct eth_fake_context *ctx = dev->driver_data;
+
+	ctx->iface = iface;
+
+	net_if_set_link_addr(iface, ctx->mac_address,
+			     sizeof(ctx->mac_address),
+			     NET_LINK_ETHERNET);
+
+	ethernet_init(iface);
+}
+
+static int eth_fake_send(struct net_if *iface,
+			 struct net_pkt *pkt)
+{
+	ARG_UNUSED(iface);
+
+	net_pkt_unref(pkt);
+
+	return 0;
+}
+
+static enum ethernet_hw_caps eth_fake_get_capabilities(struct device *dev)
+{
+	return ETH_AUTO_NEGOTIATION_SET | ETH_LINK_10BASE_T |
+		ETH_LINK_100BASE_T | ETH_DUPLEX_SET;
+}
+
+static int eth_fake_set_config(struct device *dev,
+			       enum ethernet_config_type type,
+			       const struct ethernet_config *config)
+{
+	struct eth_fake_context *ctx = dev->driver_data;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_AUTO_NEG:
+		if (config->auto_negociation == ctx->auto_negociation) {
+			return -EALREADY;
+		}
+
+		ctx->auto_negociation = config->auto_negociation;
+
+		break;
+	case ETHERNET_CONFIG_TYPE_LINK:
+		if ((config->l.link_10bt && ctx->link_10bt) ||
+		    (config->l.link_100bt && ctx->link_100bt)) {
+			return -EALREADY;
+		}
+
+		if (config->l.link_10bt) {
+			ctx->link_10bt = true;
+			ctx->link_100bt = false;
+		} else {
+			ctx->link_10bt = false;
+			ctx->link_100bt = true;
+		}
+
+		break;
+	case ETHERNET_CONFIG_TYPE_DUPLEX:
+		if (config->full_duplex == ctx->full_duplex) {
+			return -EALREADY;
+		}
+
+		ctx->full_duplex = config->full_duplex;
+
+		break;
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(ctx->mac_address, config->mac_address.addr, 6);
+
+		net_if_set_link_addr(ctx->iface, ctx->mac_address,
+				     sizeof(ctx->mac_address),
+				     NET_LINK_ETHERNET);
+		break;
+	}
+
+	return 0;
+}
+
+static struct ethernet_api eth_fake_api_funcs = {
+	.iface_api.init = eth_fake_iface_init,
+	.iface_api.send = eth_fake_send,
+
+	.get_capabilities = eth_fake_get_capabilities,
+	.set_config = eth_fake_set_config,
+};
+
+static int eth_fake_init(struct device *dev)
+{
+	struct eth_fake_context *ctx = dev->driver_data;
+
+	ctx->auto_negociation = true;
+	ctx->full_duplex = true;
+	ctx->link_10bt = true;
+	ctx->link_100bt = false;
+
+	memcpy(ctx->mac_address, mac_addr_init, 6);
+
+	return 0;
+}
+
+NET_DEVICE_INIT(eth_fake, "eth_fake", eth_fake_init, &eth_fake_data,
```

Comment:
Use ETH_NET_DEVICE_INIT() instead

Reply:
Well, we have specific macro for ethernet so why not use it. Someone will copy the code anyway for some other test :)